core/src/test/java/de/danoeh/antennapod/core/storage/DbTasksTest.java

```diff
@@ -197,6 +197,30 @@ public void testUpdateFeedRemoveUnlistedItems() {
         assertEquals(8, feedFromDB.getItems().size()); // 10 - 2 = 8 items
     }
 
+    @Test
+    public void testUpdateFeedSetDuplicate() {
+        final Feed feed = new Feed("url", null, "title");
+        feed.setItems(new ArrayList<>());
+        for (int i = 0; i < 10; i++) {
+            feed.getItems().add(
+                    new FeedItem(0, "item " + i, "id " + i, "link " + i, new Date(i), FeedItem.PLAYED, feed));
+        }
+        PodDBAdapter adapter = PodDBAdapter.getInstance();
+        adapter.open();
+        adapter.setCompleteFeed(feed);
+        adapter.close();
+
+        // change the guid of the first item, but leave the download url the same
+        FeedItem item = feed.getItemAtIndex(0);
+        item.setItemIdentifier("id 1-duplicate");
+        Feed newFeed = DBTasks.updateFeed(context, feed, false);
+        assertEquals(10, newFeed.getItems().size()); // id1-duplicate should override id 1
+
+        Feed feedFromDB = DBReader.getFeed(newFeed.getId());
+        assertEquals(10, feedFromDB.getItems().size()); // id1-duplicate should override id 1
```

Comment:
I think we should also override the guid to use the new one. Maybe it would also be good to create a download log entry that says something like "The publisher of this podcast messed up but we tried to work around it". I really think publishers should be aware of the fact that changing ids of an existing episode is bad.

Reply:
I *think* the new guid is overridden.